src/orchard/spec.clj

```diff
@@ -4,31 +4,47 @@
    [clojure.string :as str]
    [clojure.walk :as walk]))
 
-(defmacro spec [fname & args]
-  `(when-let [f# (or (resolve (symbol "clojure.spec.alpha" ~fname))
-                     (resolve (symbol "clojure.spec" ~fname)))]
-     (f# ~@args)))
-
-(defmacro spec-gen [fname & args]
-  `(when-let [f# (or (resolve (symbol "clojure.spec.gen.alpha" ~fname))
-                     (resolve (symbol "clojure.spec.gen" ~fname)))]
-     (f# ~@args)))
+(defn- spec [sym & args]
+  (when-let [f (resolve sym)]
+    (try (apply f args) (catch Exception _))))
```

Comment:
This try/catch wasn't here before and doesn't seem a great idea to me...?

Reply:
Yeah, I did this because spec-2 is not as relaxed as the others when it comes to what types of arguments you pass to it's functions. This has been moved to the individual spec fn wrappers where I tried to mirror the behavior of the current version of spec, even though we are looking up in 3 different places.